docs/spec/data-plane.md

```diff
@@ -0,0 +1,71 @@
+# Knative Eventing Data Plane Contracts
+
+## Introduction
+
+Developers using Knative eventing need to know what is supported for delivery to user provided components that receive events. Knative Eventing defines contract for data plane components and we have listed them here.
+
+## Data plane contract for Sinks
+
+A **Sink** is an [_addressable_](./interfaces.md#addressable) resource that takes responsibility for the event. A Sink could be a consumer of events, or middleware. A Sink MUST be able to receive CloudEvents over HTTP and HTTPS.
+
+A **Sink** MAY be [_callable_](./interfaces.md#callable) resource that represents an Addressable endpoint which receives an event as input and optionally returns an event to forward downstream
+
+Almost every component in Knative Eventing may be a Sink providing composability.
+
+Every Sink MUST support HTTP Protocol Binding for CloudEvents [version 1.0](https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md) and [version 0.3](https://github.com/cloudevents/spec/blob/v0.3/http-transport-binding.md).
```

Comment:
So...  for sources, we say

> Every Source SHOULD support sending events via Binary Content Mode or Structured Content Mode of the HTTP Protocol Binding for CloudEvents. Sources SHOULD send events to its Destination.

and for channel we say:

> Every Channel SHOULD support sending events via Binary Content Mode or Structured Content Mode of the HTTP Protocol Binding for CloudEvents.

Reply:
not sure on this `Every Sink MUST support HTTP Protocol Binding...`

Reply:
If it is not MUST then what is actually required for a Sink to support?